osu.Game/Overlays/KeyBinding/KeyBindingRow.cs

```diff
@@ -126,7 +127,7 @@ public void RestoreDefaults()
             {
                 var button = buttons[i++];
                 button.UpdateKeyCombination(d);
-                store.Update(button.KeyBinding);
+                store.Update((IHasGuidPrimaryKey)button.KeyBinding, k => k.KeyCombination = button.KeyBinding.KeyCombination);
```

Comment:
Not sure about this cast / data flow. Maybe feels better as a cast to `RealmKeyBinding` instead?

Reply:
would probably go for a unifying `IDatabasedKeyBinding : IKeyBinding, IHasGuidPrimaryKey` interface or something - haven't 100% checked whether it gets rid of this cast, but i can imagine it does